### ISSUE TITLE
500 Error When Using Google Integration

### ISSUE THREAD

**Issue by jimmybrancaccio** [plausible/analytics#447]
## Prerequisites
- [x] I have searched open and closed issues to make sure that the bug has not yet been reported.

## Bug report
**Describe the bug**

Attempting to connect a site in Plausible Analytics to Google results in a 500 Internal Server Error. I find the following in the logs:

```
18:46:43.168 request_id=FkZNkSl2Y3xRq2sAAABD [warn] Failed to send Sentry event.Cannot send Sentry event because of invalid DSN,
18:46:43.171 request_id=FkZNkSl2Y3xRq2sAAABD [warn] Failed to send Sentry event.Cannot send Sentry event because of invalid DSN,
18:46:43.174 [error] #PID<0.4632.0> running PlausibleWeb.Endpoint (connection #PID<0.4631.0>, stream id 1) terminated,
Server: plausible-analytics.linuxbox.ninja:80 (http),
Request: GET /auth/google/callback?state=1&code=4/0AY0e-g7M62FtUtVgaXeCzliY3c_D_nzA&scope=email%20openid%20https://www.googleapis.com/auth/userinfo.email%20https://www.googleapis.com/auth/webmasters.readonly&authuser=0&prompt=consent,
** (exit) an exception was raised:,
    ** (FunctionClauseError) no function clause matching in String.split/3,
        (elixir 1.10.3) lib/string.ex:471: String.split(nil, ".", []),
        (plausible 0.0.1) lib/plausible_web/controllers/auth_controller.ex:1: PlausibleWeb.AuthController.action/2,
        (plausible 0.0.1) lib/plausible_web/controllers/auth_controller.ex:308: PlausibleWeb.AuthController.google_auth_callback/2,
        (plausible 0.0.1) lib/plausible_web/controllers/auth_controller.ex:1: PlausibleWeb.AuthController.phoenix_controller_pipeline/2,
        (phoenix 1.4.17) lib/phoenix/router.ex:288: Phoenix.Router.__call__/2,
        (plausible 0.0.1) lib/plug/error_handler.ex:65: PlausibleWeb.Router.call/2,
        (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint.plug_builder_call/2,
        (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint."call (overridable 3)"/2
```

**Expected behavior**

I would expect there to be no error when connecting a site to Google. If it does fail, a more user-friendly error should be provided either in the web browser or the logs.

**Screenshots (If applicable)**
~If applicable, add screenshots to help explain your problem.~

**Environment (If applicable):**
 - OS: Ubuntu 20.04.x
 - Browser Safari
 - Version 14.0.1 (16610.2.11.51.8)

**Comment by ukutaht** [comment]
@jimmybrancaccio Thanks for reporting.

Did you provide `client_id` and `client_secret` as explained here? https://docs.plausible.io/self-hosting-configuration#google-search-integration

**Comment by jimmybrancaccio** [comment]
@ukutaht Thanks for the follow up!

~Yes, I provided bot the `client_id` and `client_secret`, however I've just noticed the `client_secret` does not appear to be an environment variable inside the container whereas `client_id` is. Should that be the case?~

Welp, I feel like an idiot. I was missing an `=` before the environment variable in my `docker-compose.yml` so the `client_secret` wasn't being added as an environment variable to the container... 🤦🏼‍♂️
